src/Lib/Canvas.vala

```diff
@@ -101,6 +106,38 @@ public class Akira.Lib.Canvas : Goo.Canvas {
         window.event_bus.set_focus_on_canvas.connect (on_set_focus_on_canvas);
         window.event_bus.request_escape.connect (on_set_focus_on_canvas);
         window.event_bus.insert_item.connect (on_insert_item);
+        drag_data_received.connect (on_drag_data_received);
+
+    }
+
+    // Make the Canvas Drag and Droppable
+    private void on_drag_data_received (Gdk.DragContext drag_context, int x, int y, Gtk.SelectionData data, uint info, uint time) {
+        //Loop Through The List of the dragged Files
+        int img_item_index = 0;
+        foreach (string link in data.get_uris ()) {
+            var file_link = link.replace ("file://", "").replace ("file:/", "");
+            file_link = Uri.unescape_string (file_link);
+            var image_file = File.new_for_path (file_link);
+            if (Akira.Utils.Image.is_valid_image (image_file)) {
+                var img_item_manager = new Lib.Managers.ImageManager (image_file, img_item_index);
+                window.items_manager.insert_image (img_item_manager);
+                var img_canvas_item = window.items_manager.insert_item (x, y, img_item_manager, null);
+                /*
+                 * We are creating a new Variable because when setting
+                 * img_canvas_item = (Akira.Lib.Items.CanvasImage)img_canvas_item;
+                 * when we do that it does not get converted
+                 * we want to convert it to a CanvasImage because the resize_pixbuf () function
+                */
+                Akira.Lib.Items.CanvasImage img_canvas_new_item = (Akira.Lib.Items.CanvasImage)img_canvas_item;
+                img_canvas_new_item.resize_pixbuf (-1, -1, true);
```

Comment:
Here you can simply do this:
`((Lib.Items.CanvasImage)img_canvas_item).resize_pixbuf (-1, -1, true);`
without declaring another variable.